File: src/main/java/practice/adaptor/in/web/dto/PaymentResponse.java
package practice.adaptor.in.web.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.experimental.FieldDefaults;
import practice.payment.domain.Payment;

@Getter
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class PaymentResponse {

  Long paymentId;
  String partnerPayKey;
  BigDecimal totalPayAmount;
  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
  LocalDateTime paymentDateTime;

  public static PaymentResponse of(Payment payment) {
    return PaymentResponse.builder()
        .paymentId(payment.getPaymentId())
        .partnerPayKey(payment.getPartnerPayKey())
        .totalPayAmount(payment.getTotalPayAmount())
        .paymentDateTime(payment.getPaymentDateTime())
        .build();
  }
}

File: src/main/kotlin/practice/common/dto/PaymentResponse.kt
package practice.common.dto

import com.fasterxml.jackson.annotation.JsonFormat
import practice.payment.domain.Payment
import java.math.BigDecimal
import java.time.LocalDateTime

data class PaymentResponse(
    val paymentId: Long,
    val partnerPayKey: String,
    val totalPayAmount: BigDecimal,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    val paymentDateTime: LocalDateTime
) {


    companion object {
        fun of(payment: Payment): PaymentResponse {
            return PaymentResponse(
                paymentId = payment.paymentId!!,
                partnerPayKey = payment.partnerPayKey,
                totalPayAmount = payment.totalPayAmount,
                paymentDateTime = payment.paymentDateTime!!
            )
        }
    }
}
